libs/labelbox/src/labelbox/schema/workflow/workflow.py

```diff
@@ -0,0 +1,1504 @@
+                    definition_id = WorkflowDefinitionId.InitialLabelingTask
+                    logger.warning(
+                        f"Could not determine definitionId for {NodeClass.__name__}. "
+                        f"Using {definition_id.value} as default."
+                    )
+
+        # Prepare constructor arguments with all required fields
+        raw_data = kwargs.copy()
+        constructor_args = {
+            "id": node_id,
+            "position": position,
+            "definitionId": definition_id,
+            "raw_data": raw_data,
+        }
+        constructor_args.update(kwargs)
+
+        # Create the node with all parameters
+        node = NodeClass(**constructor_args)
+
+        # Store the workflow reference in the node's raw_data
+        node.raw_data["_workflow"] = self
+
+        # Ensure we have a valid definition_id value (not Unknown)
+        if node.definition_id == WorkflowDefinitionId.Unknown:
+            logger.warning(
+                f"Node {node.id} has Unknown definition_id. "
+                f"Setting to InitialLabelingTask to prevent API errors."
+            )
+            # This is a hacky workaround since definition_id is frozen
+            # We need to modify the underlying raw_data value
+            node.raw_data["definitionId"] = (
+                WorkflowDefinitionId.InitialLabelingTask.value
+            )
+
+        # Prepare node data for config
+        node_data = {
+            "id": node.id,
+            "position": node.position.model_dump(),
+            "definitionId": (
+                WorkflowDefinitionId.InitialLabelingTask.value
+                if node.definition_id == WorkflowDefinitionId.Unknown
+                else node.definition_id.value
+            ),
+        }
+
+        # Add label if present
+        if hasattr(node, "label") and node.label:
+            node_data["label"] = node.label
+
+        # Add filterLogic if present
+        if hasattr(node, "filter_logic") and node.filter_logic:
+            node_data["filterLogic"] = node.filter_logic
+
+        # Add customFields if present
+        if hasattr(node, "custom_fields") and node.custom_fields:
+            node_data["customFields"] = node.custom_fields
+
+        # Add config if present
+        if hasattr(node, "node_config") and node.node_config:
+            node_data["config"] = node.node_config
+
+        # Add filters if present
+        if hasattr(node, "filters") and node.filters:
+            node_data["filters"] = node.filters
+
+        # Add inputs if present
+        if hasattr(node, "inputs") and node.inputs:
+            node_data["inputs"] = node.inputs
+
+        # Add to config
+        self.config["nodes"].append(node_data)
+
+        # Reset the nodes cache to ensure it's up-to-date
+        self._nodes_cache = None
+
+        return node
+
+    # Node creation methods
+    def InitialLabelingNode(self, **kwargs) -> InitialLabelingNode:
```

Comment:
Personally, I prefer `create_initial_labeling_node` or verb-based method names - this is a bit unorthodox but it's fine if this a convention used by the SDK.

Reply:
Code refactored